lib/block_view.go

```diff
@@ -2919,24 +3053,72 @@ func _isEntryImmatureBlockReward(utxoEntry *UtxoEntry, blockHeight uint32, param
 	return false
 }
 
-func _verifySignature(txn *MsgBitCloutTxn) error {
-	// Compute a hash of the transaction
+func (bav *UtxoView) _verifySignature(txn *MsgBitCloutTxn, blockHeight uint32) error {
+	// Compute a hash of the transaction.
 	txBytes, err := txn.ToBytes(true /*preSignature*/)
 	if err != nil {
 		return errors.Wrapf(err, "_verifySignature: Problem serializing txn without signature: ")
 	}
 	txHash := Sha256DoubleHash(txBytes)
-	// Convert the txn public key into a *btcec.PublicKey
-	txnPk, err := btcec.ParsePubKey(txn.PublicKey, btcec.S256())
+	sig, err := btcec.ParseDERSignature(txn.Signature, btcec.S256())
+
+	// Look for the derived key in transaction ExtraData and validate it. For transactions
+	// signed using a derived key, the derived public key is passed to ExtraData.
+	var derivedPk *btcec.PublicKey
+	var derivedPkBytes []byte
+	isDerived := false
+	if txn.ExtraData != nil {
+		if derivedPkBytes, isDerived = txn.ExtraData[DerivedPublicKey]; isDerived {
```

Comment:
Nit: Avoid assignment in if statements since this can be error-prone. Instead prefer:

```
derivedPkBytes, isDerived = txn.ExtraData[DerivedPublicKey];
if isDerived {
...
}
```